front/ui/ui-charts/src/manchette/hooks/useManchetteWithSpaceTimeChart.tsx

```diff
@@ -643,18 +675,20 @@ const useManchetteWithSpaceTimeChart = ({
 
             const newState = { ...prev };
             const { initialOffset } = panning;
-            newState.xOffset = initialOffset.x + diff.x;
-
-            const newYPos = initialOffset.y - diff.y;
-            if (
-              manchetteWithSpaceTimeChartRef.current &&
-              newYPos >= 0 &&
-              newYPos + manchetteWithSpaceTimeChartRef.current.offsetHeight <
-                manchetteWithSpaceTimeChartRef.current.scrollHeight
-            ) {
-              newState.yOffset = newYPos;
-              manchetteWithSpaceTimeChartRef.current.scrollTop = newYPos;
+            const manchette = manchetteWithSpaceTimeChartRef.current;
+
+            if (enableTimePan) {
+              newState.xOffset = initialOffset.x + diff.x;
             }
+            if (enableSpacePan) {
+              let newYOffset = initialOffset.y - diff.y;
+              newYOffset = Math.max(newYOffset, 0);
+              if (manchette)
+                newYOffset = Math.min(newYOffset, manchette.scrollHeight - manchette.offsetHeight);
+              newState.yOffset = newYOffset;
+              if (manchette) manchette.scrollTop = newYOffset;
```

Comment:
nit: only check manchette once
```suggestion
              if (manchette){
                newYOffset = Math.min(newYOffset, manchette.scrollHeight - manchette.offsetHeight);
                manchette.scrollTop = newYOffset;
              }
              newState.yOffset = newYOffset;
```